### PR TITLE
feat: event debug view in the toolbar

### PR DESCRIPTION
**Pull request by pauldambra**
A long walk along the coast of trapani had lots of ideas... this was one I could make while recovering from the walk

(needs https://github.com/PostHog/posthog-js/pull/1136)

Screenshot 2024-04-13 at 17 49 34
Screenshot 2024-04-13 at 17 49 47

### CODE REVIEW

File: frontend/src/toolbar/debug/EventDebugMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                        <LemonSwitch
+                            checked={showRecordingSnapshots}
+                            onChange={(c) => setShowRecordingSnapshots(c)}
+                            label="Show recording snapshot events"
+                        />
+                    </div>
+                </div>
+            </ToolbarMenu.Header>
+            <ToolbarMenu.Body>
+                <div className="flex flex-col space-y-1 LiveCapture">
```
Comment (benjackwhite) [inline frontend/src/toolbar/debug/EventDebugMenu.tsx:38]:
```suggestion
                <div className="flex flex-col space-y-1">
```

File: frontend/src/toolbar/debug/EventDebugMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    const { events, isCollapsedEventRow, expandedEvent, showRecordingSnapshots } = useValues(eventDebugMenuLogic)
+    const { markExpanded, setShowRecordingSnapshots } = useActions(eventDebugMenuLogic)
+
+    return (
+        <ToolbarMenu>
+            <ToolbarMenu.Header>
+                <div className="flex flex-col pb-2 space-y-1">
+                    <div className="flex flex-row justify-around">
+                        <span className="text-xs">
+                            Seen {events.filter((e) => e.event !== '$snapshot').length} events.
```
Comment (benjackwhite) [inline frontend/src/toolbar/debug/EventDebugMenu.tsx:22]: Nit. This should be in the logic. Especially as the list gets bigger and bigger.

File: frontend/src/toolbar/debug/EventDebugMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                            onChange={(c) => setShowRecordingSnapshots(c)}
+                            label="Show recording snapshot events"
+                        />
+                    </div>
+                </div>
+            </ToolbarMenu.Header>
+            <ToolbarMenu.Body>
+                <div className="flex flex-col space-y-1 LiveCapture">
+                    {events
+                        .filter((e) => {
```
Comment (benjackwhite) [inline frontend/src/toolbar/debug/EventDebugMenu.tsx:40]: Move to the logic?

**Review by benjackwhite — CHANGES_REQUESTED** [review]

File: frontend/src/toolbar/debug/EventDebugMenu.tsx (hunk 2 above)
Comment (pauldambra) [reply frontend/src/toolbar/debug/EventDebugMenu.tsx:22]: My inner Ben even said this to me when I was adding it 🤣 I should have listened to them

**Review by benjackwhite — APPROVED** [review]

**Pull request merged**